statsmodels/tsa/statespace/_initialization.pyx.in

```diff
@@ -0,0 +1,248 @@
+    cdef int clear_cov(self, int offset, {{cython_type}} [::1, :] cov) except 1:
+        cov[offset:offset + self.k_states, offset:offset + self.k_states] = 0
+        return 0
+
+    cdef int initialize_known_constant(self, int offset,
+                                        {{cython_type}} [:] initial_state_mean) except 1:
+        cdef int inc = 1
+        blas.{{prefix}}copy(&self.k_states, &self.constant[0], &inc,
+                                     &initial_state_mean[offset], &inc)
+
+        return 0
+
+    cdef int initialize_known_stationary_cov(self, int offset,
+                                              {{cython_type}} [::1, :] initial_stationary_state_cov) except 1:
+        cdef int i, inc = 1
+        # Copy columns
+        for i in range(self.k_states):
+            blas.{{prefix}}copy(&self.k_states, &self.stationary_cov[0, i], &inc,
+                                         &initial_stationary_state_cov[offset, offset + i], &inc)
+
+        return 0
+
+    cdef int initialize_diffuse(self, int offset,
+                                 {{cython_type}} [::1, :] initial_diffuse_state_cov) except 1:
+        cdef int i
+        for i in range(offset, offset + self.k_states):
+            initial_diffuse_state_cov[i, i] = 1
+
+        return 0
+
+    cdef int initialize_approximate_diffuse(self, int offset,
+                                             {{cython_type}} [::1, :] initial_stationary_state_cov) except 1:
+        cdef int i
+        for i in range(offset, offset + self.k_states):
+            initial_stationary_state_cov[i, i] = self.approximate_diffuse_variance
+
+        return 0
+
+    cdef int initialize_stationary_constant(self, int offset, {{prefix}}Statespace model,
+                                    {{cython_type}} [:] initial_state_mean,
+                                    int complex_step=False) except 1:
+
+        cdef:
+            np.npy_intp dim2[2]
+            int i, info, inc = 1
+            int k_states2 = self.k_states**2
+            np.float64_t asum, tol = 1e-9
+            cdef {{cython_type}} scalar
+            cdef int [::1,:] ipiv
+
+        # Clear the unconditional mean (for this block)
+        initial_state_mean[offset:offset + self.k_states] = 0
+
+        # Check if the state intercept is all zeros; if it is, then the
+        # unconditional mean is also all zeros
+        {{if combined_prefix == 'd'}}
+        asum = blas.{{prefix}}asum(&model.k_states, &model.state_intercept[0, 0], &inc)
+        {{elif prefix == 'c'}}
+        asum = blas.scasum(&model.k_states, &model.state_intercept[0, 0], &inc)
+        {{else}}
+        asum = blas.dzasum(&model.k_states, &model.state_intercept[0, 0], &inc)
+        {{endif}}
+
+        # If the state intercept is non-zero, compute the mean
+        if asum > tol:
+            dim2[0] = self.k_states
+            dim2[1] = self.k_states
+            ipiv = np.PyArray_ZEROS(2, dim2, np.NPY_INT32, FORTRAN)
+
+            # Create T - I
```

Comment:
In the current version of representation.pyx this is `# I - T`.  Why the change?

Reply:
I guess it's just easier to do using the BLAS calls that I know (of course I'm not ruling out that there may be a better way to do it altogether): subtracting 1 from the diagonal elements and then scaling the whole thing by -1 is faster than constructing an identity matrix and subtracting the T matrix from it.